tests/test_dataprep_pgvector.sh

```diff
@@ -21,20 +24,72 @@ function start_service() {
 
     sleep 3m
 }
 
 function validate_microservice() {
-    URL="http://$ip_address:6007/v1/dataprep"
-    echo 'The OPEA platform includes: Detailed framework of composable building blocks for state-of-the-art generative AI systems including LLMs, data stores, and prompt engines' > ./dataprep_file.txt
-    curl --noproxy $ip_address --location --request POST \
-      --form 'files=@./dataprep_file.txt' $URL
+    cd $LOG_PATH
+
+    # test /v1/dataprep
+    URL="http://${ip_address}:$dataprep_service_port/v1/dataprep"
+    echo "Deep learning is a subset of machine learning that utilizes neural networks with multiple layers to analyze various levels of abstract data representations. It enables computers to identify patterns and make decisions with minimal human intervention by learning from large amounts of data." > $LOG_PATH/dataprep_file.txt
+    HTTP_STATUS=$(curl -s -o /dev/null -w "%{http_code}" -X POST -F 'files=@./dataprep_file.txt' -H 'Content-Type: multipart/form-data' "$URL")
+    if [ "$HTTP_STATUS" -eq 200 ]; then
+        echo "[ dataprep ] HTTP status is 200. Checking content..."
+        cp ./dataprep_file.txt ./dataprep_file2.txt
+        local CONTENT=$(curl -s -X POST -F 'files=@./dataprep_file2.txt' -H 'Content-Type: multipart/form-data' "$URL" | tee ${LOG_PATH}/dataprep.log)
+
+        if echo "$CONTENT" | grep -q "Data preparation succeeded"; then
+            echo "[ dataprep ] Content is as expected."
+        else
+            echo "[ dataprep ] Content does not match the expected result: $CONTENT"
+            docker logs dataprep-pgvector >> ${LOG_PATH}/dataprep.log
+            exit 1
+        fi
+    else
+        echo "[ dataprep ] HTTP status is not 200. Received status was $HTTP_STATUS"
+        docker logs dataprep-pgvector >> ${LOG_PATH}/dataprep.log
+        exit 1
+    fi
+
+    # test /v1/dataprep/get_file
+    URL="http://${ip_address}:$dataprep_service_port/v1/dataprep/get_file"
+    HTTP_STATUS=$(curl -s -o /dev/null -w "%{http_code}" -X POST -H 'Content-Type: application/json' "$URL")
+    if [ "$HTTP_STATUS" -eq 200 ]; then
+        echo "[ dataprep - file ] HTTP status is 200. Checking content..."
+        local CONTENT=$(curl -s -X POST -H 'Content-Type: application/json' "$URL" | tee ${LOG_PATH}/dataprep_file.log)
+
+        if echo "$CONTENT" | grep -q '{"name":'; then
+            echo "[ dataprep - file ] Content is as expected."
+        else
+            echo "[ dataprep - file ] Content does not match the expected result: $CONTENT"
+            docker logs dataprep-pgvector >> ${LOG_PATH}/dataprep_file.log
+            exit 1
+        fi
+    else
+        echo "[ dataprep - file ] HTTP status is not 200. Received status was $HTTP_STATUS"
+        docker logs dataprep-pgvector >> ${LOG_PATH}/dataprep_file.log
+        exit 1
+    fi
+
+    # test /v1/dataprep/delete_file
+    URL="http://${ip_address}:$dataprep_service_port/v1/dataprep/delete_file"
```

Comment:
could you please check https://github.com/opea-project/GenAIComps/issues/394 to see if it's ok to use DELETE method but not different endpoints to do this operation?

Reply:
I believe this issue is still open and changes are still WIP. 
Will update the component once the feature support for GET/DELETE is available in the mainline.

Reply:
ok